uint/src/uint.rs

```diff
@@ -1550,31 +1550,34 @@ macro_rules! construct_uint {
 			}
 		}
 
-		$crate::impl_std_for_uint!($name, $n_words);
-		// `$n_words * 8` because macro expects bytes and
-		// uints use 64 bit (8 byte) words
-		$crate::impl_quickcheck_arbitrary_for_uint!($name, ($n_words * 8));
-		$crate::impl_arbitrary_for_uint!($name, ($n_words * 8));
-	}
-}
-
-#[cfg(feature = "std")]
-#[macro_export]
-#[doc(hidden)]
-macro_rules! impl_std_for_uint {
-	($name: ident, $n_words: tt) => {
 		impl $crate::core_::str::FromStr for $name {
-			type Err = $crate::rustc_hex::FromHexError;
+			type Err = $crate::hex::FromHexError;
```

Comment:
changing a pub type is a breaking change...
maybe we can change it in a way that doesn't tie `hex`'s version to this lib's version

Reply:
I could transform hex's error into rustc_hex one - but that would really be crutches. If we are going to use hex long-term anyway, we might as well use its error type.

Reply:
I was thinking of using an opaque error type that just impl `std::error::Error` and doesn't expose third-party library types

Reply:
But that would be copy-pasting `hex::FromHexError`, as the latter is pretty minimal already. 🙂

Reply:
the point is we woudn't need to bump version on `hex` bump or if we replace it with another lib

Reply:
and by opaque I don't mean a `pub enum`, but
```rust
pub struct Error {
/* private fields */
}
```

Reply:
But macro will be used outside `uint` crate, so either the Error struct must have public constructor, or each macro must create a novel error type.

One option is to just erase error type and use `anyhow::Error` here.

Reply:
yeah, I guess we could have a public construction (or a `From` impl) with `#[doc(hidden)]` as we do with `hex` reexport.

Reply:
done